exec/plugin-commands-script-runners/src/dlx.ts

```diff
@@ -190,17 +193,28 @@ export function createCacheKey (pkgs: string[], registries: Record<string, strin
   return createBase32Hash(hashStr)
 }
 
-function isCacheValid (cacheLink: string, dlxCacheMaxAge: number): boolean {
+function getValidCacheDir (cacheLink: string, dlxCacheMaxAge: number): string | undefined {
   let stats: Stats
+  let target: string
   try {
     stats = fs.lstatSync(cacheLink)
+    if (stats.isSymbolicLink()) {
+      target = fs.readlinkSync(cacheLink)
+      if (!target) return undefined
+    } else {
+      return undefined
+    }
   } catch (err) {
     if (util.types.isNativeError(err) && 'code' in err && err.code === 'ENOENT') {
-      return false
+      return undefined
     }
     throw err
   }
-  return stats.mtime.getTime() + dlxCacheMaxAge * 60_000 >= new Date().getTime()
+  const isValid = stats.mtime.getTime() + dlxCacheMaxAge * 60_000 >= new Date().getTime()
+  if (isValid) {
+    return path.resolve(path.dirname(cacheLink), target)
```

Comment:
Assuming you applied the above suggestion,

```suggestion
    return target
```